### ISSUE TITLE
ssl.SSLCertVerificationError: [SSL: CERTIFICATE_VERIFY_FAILED] - Unable to connect to HA URL

### ISSUE THREAD

**Issue by 007-DJ** [custom-components/alexa_media_player#1648]
Description:
Previously, I was never able to get the HA built-in F2A to work as expected; consequently, I need to re-auth approximately weekly. I am not able to recall the previous success methods as they all had varied until initialization success. Here are the steps I am attempting:

1. Removed {#} alexa_media entries from configuration.yaml. Restart HA.
2. Removed alexa media player from HACS. Restart HA.
3. Enabled alexa_media entries in configuration.yaml. Restart HA.
4. Configure (Discovered) Alexa Media Player in integrations.
5. Verified prefilled fields are all correct: EMail Address, Password, duckdns.org link, Built-in F2A App Key (Seed) from Amazon F2A page for adding second S2V authenticator:
![3](https://user-images.githubusercontent.com/98818663/174824267-cfcc5c39-c594-4c95-afda-9e80b88c3be2.JPG)
With repeated efforts, I would get one of two errors: Unknown error occurred or unable to connect.
![4](https://user-images.githubusercontent.com/98818663/174825179-2f95924f-8399-4367-9bfd-91be1232a096.JPG)
![2](https://user-images.githubusercontent.com/98818663/174825276-c3510195-1603-4ad9-a1b3-b82fef0aa291.JPG)
For clarification purposes, the https address for HA in duckdns.org is verified by coping it into a browser and opening HA.

**System details**
Home Assistant Core 2022.6.6
Home Assistant Supervisor 2022.05.3
Home Assistant OS 8.2
Alexa Media Player v4.0.2 located in '/config/custom_components/alexa_media'
Path to configuration.yaml: /config/configuration.yaml
 - Amazon [2FA is enabled](https://github.com/custom-components/alexa_media_player/wiki/Configuration#enable-two-factor-authentication-for-your-amazon-account) (yes).
 
**Logs**
home-assistant.log:
2022-06-21 09:28:24 WARNING (SyncWorker_0) [homeassistant.loader] We found a custom integration alexa_media which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant
2022-06-21 09:28:24 WARNING (SyncWorker_1) [homeassistant.loader] We found a custom integration hacs which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant
2022-06-21 09:28:31 ERROR (MainThread) [homeassistant.components.notify] Error setting up platform smtp
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/components/notify/legacy.py", line 67, in async_setup_platform
    notify_service = await hass.async_add_executor_job(
  File "/usr/local/lib/python3.9/concurrent/futures/thread.py", line 58, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/usr/src/homeassistant/homeassistant/components/smtp/notify.py", line 92, in get_service
    if mail_service.connection_is_valid():
  File "/usr/src/homeassistant/homeassistant/components/smtp/notify.py", line 154, in connection_is_valid
    server = self.connect()
  File "/usr/src/homeassistant/homeassistant/components/smtp/notify.py", line 144, in connect
    mail.starttls(context=ssl_context)
  File "/usr/local/lib/python3.9/smtplib.py", line 790, in starttls
    self.sock = context.wrap_socket(self.sock,
  File "/usr/local/lib/python3.9/ssl.py", line 500, in wrap_socket
    return self.sslsocket_class._create(
  File "/usr/local/lib/python3.9/ssl.py", line 1040, in _create
    self.do_handshake()
  File "/usr/local/lib/python3.9/ssl.py", line 1309, in do_handshake
    self._sslobj.do_handshake()
ssl.SSLCertVerificationError: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: Hostname mismatch, certificate is not valid for 'mail.myfairpoint.net'. (_ssl.c:1129)
2022-06-21 09:28:31 ERROR (MainThread) [homeassistant.components.notify] Error setting up platform smtp
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/components/notify/legacy.py", line 67, in async_setup_platform
    notify_service = await hass.async_add_executor_job(
  File "/usr/local/lib/python3.9/concurrent/futures/thread.py", line 58, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/usr/src/homeassistant/homeassistant/components/smtp/notify.py", line 92, in get_service
    if mail_service.connection_is_valid():
  File "/usr/src/homeassistant/homeassistant/components/smtp/notify.py", line 154, in connection_is_valid
    server = self.connect()
  File "/usr/src/homeassistant/homeassistant/components/smtp/notify.py", line 144, in connect
    mail.starttls(context=ssl_context)
  File "/usr/local/lib/python3.9/smtplib.py", line 790, in starttls
    self.sock = context.wrap_socket(self.sock,
  File "/usr/local/lib/python3.9/ssl.py", line 500, in wrap_socket
    return self.sslsocket_class._create(
  File "/usr/local/lib/python3.9/ssl.py", line 1040, in _create
    self.do_handshake()
  File "/usr/local/lib/python3.9/ssl.py", line 1309, in do_handshake
    self._sslobj.do_handshake()
ssl.SSLCertVerificationError: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: Hostname mismatch, certificate is not valid for 'mail.myfairpoint.net'. (_ssl.c:1129)
2022-06-21 10:17:16 WARNING (SyncWorker_6) [homeassistant.loader] We found a custom integration hacs which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant

The Alexa Media Player integration is the backbone of my HA. Any assistance is greatly apricated. Please let me know How I can return the favor.

**Comment by MitkaR88** [comment]
Ditto here. 
Region Amazon.com

**Comment by Albruaje** [comment]
Same for me, but with amazon.es

**Comment by kevinwells22** [comment]
It seems the '2FA Code' option box has dissapeared in this release, now we only have the 'Built-in 2FA App key' option, which is not for use with a generated key from Authenticator.

If you leave that box empty and hit 'Submit' it takes you to an amazon page where you can enter the 2FA key, and all works well, Alexa Medai Player is now authenticaed and working well, but if I restart Homeassistant, then it need re-authenticating all over again.

Anyone on here from the dev team can point us in the right direction please?

**Comment by alandtse** [comment]
> ssl.SSLCertVerificationError: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: Hostname mismatch, certificate is not valid for 'mail.myfairpoint.net'. (_ssl.c:1129)

There's a SSL host mismatch. Integration is trying to test the url you provide from the HA machine. If for some reason the cert fails, it will give that error. Use the local IP as a workaround or correct your cert as exposed to your HA machine.

Everyone else, if you're not talking about a SSL cert issue, you're off topic. Please search as almost everything asked has been already discussed and closed. Please open a new issue if your issue is actually new.

**Comment by 007-DJ** [comment]
@alandtse 
I simply can not express how much your direction is appreciated. I wanted to share my work-around. You mentioned correcting my SSL cert (Note: Using HACS add-on NGINX Home Assistant SSL Proxy). Not knowing how, I reinstalled using my local http address. This gave me the success But could not be loaded message:
![image](https://user-images.githubusercontent.com/98818663/175509263-0aa05392-fdca-4025-bb9e-1deec1e6ba9e.png)
![image](https://user-images.githubusercontent.com/98818663/175509735-1ecfb809-c02e-407d-a8f1-93fcdfacbbb8.png)
My workaround Was to delete and reload with the exact same instructions, but this time on the Amazon Login Authentication Screen, I enabled the option to not require F2P and it complete the integration.
As mentioned, This is the most important HA integration. without it, HA is nothing more than a status app.
THANK YOU!!!
